### PR TITLE
Initialize DB when VATZ is initialized.

### LINKED ISSUE #421: Initialize database
## Checklist

- [x] New Feature for the Service/Plugin (Vatz)
- [ ] Enhancement (Vatz)
- [ ] others(etc. e.g, documentation, project policy management)

---

`vatz init` command provides initialize process to make VATZ prepared to start.
For now, it only creates the default config file, so it is time to create the default database file also.

### PR DESCRIPTION
**Pull request by rootwarp**
### 1. Type of change

Please delete options that are not relevant.

- [x] New feature
- [ ] Enhancement
- [ ] Bug/fix (non-breaking change which fixes an issue)
- [ ] others (anything other than above)

---

**Related:** # (issue)

close #421 

**Summary**

Initialize DB when `vatz init` executed.

---

### CODE REVIEW

**Comment by rootwarp** [comment]
```
ok      github.com/dsrvlabs/vatz/utils  (cached)
?       github.com/dsrvlabs/vatz        [no test files]
ok      github.com/dsrvlabs/vatz/cmd    3.137s  coverage: 19.1% of statements
ok      github.com/dsrvlabs/vatz/manager/api    0.501s  coverage: 0.0% of statements [no tests to run]
ok      github.com/dsrvlabs/vatz/manager/config 0.268s  coverage: 91.2% of statements
ok      github.com/dsrvlabs/vatz/manager/dispatcher     0.894s  coverage: 7.8% of statements
ok      github.com/dsrvlabs/vatz/manager/executor       1.144s  coverage: 68.9% of statements
ok      github.com/dsrvlabs/vatz/manager/healthcheck    0.628s  coverage: 46.5% of statements
ok      github.com/dsrvlabs/vatz/manager/plugin 7.026s  coverage: 67.4% of statements
?       github.com/dsrvlabs/vatz/manager/types  [no test files]
?       github.com/dsrvlabs/vatz/mocks  [no test files]
ok      github.com/dsrvlabs/vatz/monitoring/prometheus  1.300s  coverage: 0.0% of statements
ok      github.com/dsrvlabs/vatz/rpc    2.632s  coverage: 67.2% of statements
ok      github.com/dsrvlabs/vatz/utils  1.671s  coverage: 13.3% of statements
go build -ldflags="-X 'github.com/dsrvlabs/vatz/utils.Version=issue#421' -X 'github.com/dsrvlabs/vatz/utils.Commit=a06a1fe'" -v
rootwarp@Mindhack2021 ~/g/d/vatz issue#421 >
```

File: manager/plugin/db.go [hunk 1]
```diff
@@ -246,3 +250,37 @@ func getDBConnection(ctx context.Context, dbfile string) (*sql.Conn, error) {
...
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		_ = os.Mkdir(path, 0755)
+	}
+
+	_, err = newWriter(dbfile)
+	if err != nil {
+		return err
+	}
+
+	err = db.createPluginTable()
```
[the remark this answers is not in the document]
Comment (rootwarp) [reply manager/plugin/db.go:274]: @heejin-github Thanks for the comment.
I removed duplicated table creation at 6755499.

**Review by heejin-github — APPROVED** [review]
LGTM

**Comment by xellos00** [comment]
@rootwarp 
I think I should have catch this error, 
but is this command shouldn't valid for when I init to other directory? 

or we skipped --home command? 

```
 ./vatz init --home /Users/dongyookang/test
Error: unknown flag: --home
Usage:
   init [flags]

Flags:
  -h, --help            help for init
  -o, --output string   New config file to create (default "default.yaml")

panic: unknown flag: --home

goroutine 1 [running]:
main.main()
        /Users/dongyookang/ffplay/GolandProjects/dsrvlabs/vatz/main.go:20 +0x3a
 ✘ dongyookang DK 🚀   ~/ffplay/GolandProjects/dsrvlabs/vatz   main ✚
 

```

**Comment by rootwarp** [comment]
> @rootwarp I think I should have catch this error, but is this command shouldn't valid for when I init to other directory?
> 
> or we skipped --home command?
> 
> ```
>  ./vatz init --home /Users/dongyookang/test
> Error: unknown flag: --home
> Usage:
>    init [flags]
> 
> Flags:
>   -h, --help            help for init
>   -o, --output string   New config file to create (default "default.yaml")
> 
> panic: unknown flag: --home
> 
> goroutine 1 [running]:
> main.main()
>         /Users/dongyookang/ffplay/GolandProjects/dsrvlabs/vatz/main.go:20 +0x3a
>  ✘ dongyookang DK 🚀   ~/ffplay/GolandProjects/dsrvlabs/vatz   main ✚
>  
> ```

@xellos00 Thanks for the review.
`--home` flag should be available, so I'll be back with a new commit. 👍

**Comment by rootwarp** [comment]
@xellos00 I'm considering for adding `--home` flag, but we need to discuss more before implementing it.

When we try to execute `vatz start` command, we can use `--config` flag to choose the exact configuration file. So, if we try to make the home directory selectable, the selected directory should be written on the config file and `vatz` should load home directory and use it to load DB and configs.

Moreover, if we add `home` field on the config file, we may update other commands because those are just using the VATZ home directory as `$HOME/.vatz` implicitly.

So, whole steps I expected are,
1. Add `--home` flag on `init` command. The `init` command initializes DB and creates new configs. And it also adds `home` field on the new configs.
2. Update `plugin` command to load `home` directory from config.
3. Update `start` command to load `home` directory from config.

**Comment by xellos00** [comment]
> @xellos00 I'm considering for adding `--home` flag, but we need to discuss more before implementing it.
> 
> When we try to execute `vatz start` command, we can use `--config` flag to choose the exact configuration file. So, if we try to make the home directory selectable, the selected directory should be written on the config file and `vatz` should load home directory and use it to load DB and configs.
> 
> Moreover, if we add `home` field on the config file, we may update other commands because those are just using the VATZ home directory as `$HOME/.vatz` implicitly.
> 
> So, whole steps I expected are,
> 
> 1. Add `--home` flag on `init` command. The `init` command initializes DB and creates new configs. And it also adds `home` field on the new configs.
> 2. Update `plugin` command to load `home` directory from config.
> 3. Update `start` command to load `home` directory from config.

In fact, I understood your expected behaviors regarding `--home` flag in command `init, start`. 
yet I am little bit confused after I read your comment. 

let's have further discussion tomorrow for this. 
I sent invitation as below. 
image

**Comment by xellos00** [comment]
Confirmed!! 

```
 dongyookang DK 🚀   ~/ffplay/GolandProjects/dsrvlabs/vatz   issue#421 ±✚
 ./vatz init       
2023-04-11T06:35:58-05:00 INF init module=main
2023-04-11T06:35:58-05:00 INF create file default.yaml module=main
2023-04-11T06:35:58-05:00 INF initDB /Users/dongyookang/.vatz/vatz.db module=db
2023-04-11T06:35:58-05:00 INF newWriter /Users/dongyookang/.vatz/vatz.db module=db
2023-04-11T06:35:58-05:00 INF Create DB Instance module=db
 dongyookang DK 🚀   ~/ffplay/GolandProjects/dsrvlabs/vatz   issue#421 ±✚
 ./vatz plugin list
2023-04-11T06:36:12-05:00 INF List plugins module=plugin
2023-04-11T06:36:12-05:00 INF List module=plugin
2023-04-11T06:36:12-05:00 INF newReader /Users/dongyookang/.vatz/vatz.db module=db
2023-04-11T06:36:12-05:00 INF Create DB Instance module=db
2023-04-11T06:36:12-05:00 INF List Plugin module=db
+------+--------------+------------+---------+
| NAME | INSTALL DATA | REPOSITORY | VERSION |
+------+--------------+------------+---------+
+------+--------------+------------+---------+
 dongyookang DK 🚀   ~/ffplay/GolandProjects/dsrvlabs/vatz   issue#421 ±✚
 

```



```
 dongyookang DK 🚀   ~
 ls -al | grep .vatz
drwxr-xr-x    3 dongyookang  staff      96 Apr 11 06:26 .vatz
drwxr-xr-x    5 dongyookang  staff     160 Dec  1 22:11 .vatz_backup
 dongyookang DK 🚀   ~
 rm -rf .vatz     
 dongyookang DK 🚀   ~
 ls -al | grep .vatz
drwxr-xr-x    5 dongyookang  staff     160 Dec  1 22:11 .vatz_backup
 dongyookang DK 🚀   ~
 ls -al | grep .vatz
drwxr-xr-x    3 dongyookang  staff      96 Apr 11 06:35 .vatz
drwxr-xr-x    5 dongyookang  staff     160 Dec  1 22:11 .vatz_backup
 dongyookang DK 🚀   ~
 

```

**Review by xellos00 — APPROVED** [review]

**Pull request merged**